internal/controller/apiextensions/claim/syncer_ssa.go

```diff
@@ -74,32 +72,68 @@ func NewPatchingManagedFieldsUpgrader(w client.Writer) *PatchingManagedFieldsUpg
 
 // Upgrade the supplied object's field managers from client-side to server-side
 // apply.
-func (u *PatchingManagedFieldsUpgrader) Upgrade(ctx context.Context, obj client.Object, ssaManager string, csaManagers ...string) error {
-	// UpgradeManagedFieldsPatch removes or replaces the specified CSA managers.
-	// Unfortunately most Crossplane controllers use CSA manager "crossplane".
-	// So we could for example fight with the XR controller:
-	//
-	// 1. We remove CSA manager "crossplane", triggering XR controller watch
-	// 2. XR controller uses CSA manager "crossplane", triggering our watch
-	// 3. Back to step 1 :)
-	//
-	// In practice we only need to upgrade once, to ensure we don't share fields
-	// that only this controller has ever applied with "crossplane". We assume
-	// that if our SSA manager already exists, we've done the upgrade.
-	for _, e := range obj.GetManagedFields() {
+//
+// This is a multi-step process.
+//
+// Step 1: All fields are owned by manager 'crossplane' operation 'Update'. This
```

Comment:
nit: this isn't entirely accurate as XRs based on Composition functions would have other ssa based managers, right ?